python/pyspark/sql/connect/expressions.py

```diff
@@ -438,6 +438,59 @@ def __repr__(self) -> str:
             return f"{self._name}({', '.join([str(arg) for arg in self._args])})"
 
 
+class PythonFunction(Expression):
```

Comment:
Using `PythonFunction` as a subclass of `Expression` makes only sense if it can be used as an individual expression. Since this is not the case, I would not define it as such. This gives you the freedom to essentially use the same interface but return `PythonFunction` in the `to_plan` method.

In addition, it might be questionable to create such a large wrapper for just a byte array.

Reply:
Makes sense! Since `PythonUDF` is not used as an individual expression after introducing `ScalarInlineUserDefinedFunction`, it doesn't extend `Expression` either.